ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -88,8 +88,24 @@ const STACK_DIRECTION: StackDirection = StackDirection::GrowsDown;
 #[inline(never)]
 pub extern "C" fn __yk_break() {}
 
+/// A simple front end for the X86_64 code generator.
+pub(crate) struct X64CodeGenFrontEnd;
```

Comment:
Maybe a bikeshed too far, but this name confused me a bit, because a code generator is itself a backend. I know what you mean though, this is the frontend of the backend, right?

Reply:
This is a minimal hack to nudge us in the right direction: the names aren't good, but the structure isn't quite right either. We can gradually improve things in future PRs.